libraries/azure_generic_resources.rb

```diff
@@ -0,0 +1,179 @@
+    end
+  EXAMPLE
+
+  attr_reader :table
+
+  def initialize(opts = {}, static_resource = false)
+    # A HTTP client will be created in the backend.
+    super(opts)
+    # This will be false in the base class if AzureConnection fails.
+    return unless http_client_ready?
+
+    @display_name = @opts.slice(:resource_group, :resource_path, :name, :resource_provider, :tag_name, :tag_value).values.join(' ')
+    if static_resource
+      raise ArgumentError, 'Warning for the resource author: `resource_provider` must be defined.' \
+      unless opts.key?(:resource_provider)
+      @table = []
+      @resources = {}
+      opts[:api_version] = 'latest' unless opts.key?(:api_version)
+      # These are the parameters created in the static resource code, NOT provided by the user.
+      allowed_params = %i(resource_path resource_group resource_provider)
+      # User provided parameters will be passed here for validation with:
+      #   opts[:required_parameters]
+      #   opts[:allowed_parameters]
+      allowed_params += opts[:allowed_parameters] unless opts[:allowed_parameters].nil?
+      parameters_to_validate = {
+        required: opts[:required_parameters],
+      allow: allowed_params,
+      }.each_with_object({}) { |(k, v), acc| acc[k] = v unless v.nil? }
+      validate_parameters(**parameters_to_validate)
+      @display_name = @opts[:display_name] unless @opts[:display_name].nil?
+      return
+    end
+
+    # Either one of the following sets can be provided for a valid short description query.
+    # resource_group
+    # name
+    # substring_of_name, substring_of_resource_group
+    # tag_name + tag_value
+    # resource_group + resource_provider
+    raise ArgumentError, "#{@__resource_name__}: The `api_version` parameter is not allowed." if opts.key?(:api_version)
+    validate_parameters(allow: %i(name
+                                  substring_of_name
+                                  resource_group
+                                  substring_of_resource_group
+                                  resource_provider
+                                  tag_name
+                                  tag_value
+                                  location))
+    filter = @opts.slice(:resource_group, :name, :resource_provider, :tag_name, :tag_value, :location,
+                         :substring_of_name, :substring_of_resource_group)
+    catch_failed_resource_queries do
+      # This filter will be used to query the Rest API.
+      # At this point the resource_provider should be identical to resource_type which is the allowed query parameter.
+      filter[:resource_type] = filter[:resource_provider] unless filter[:resource_provider].nil?
+      filter.delete(:resource_provider)
+      @resources = resource_short(filter)
+    end
+    # If an exception is raised above then the resource is failed.
+    # This check should be done every time after using catch_failed_resource_queries
+    return if failed_resource?
```

Comment:
I'm wondering whether this part of `catch_failed_resource_queries` is sufficient for this to fail gracefully:
```resource_fail(message)```
as this will bubble the fail up to Inspec itself `fail_resource(message)`

I was under the impression `fail_resource` would cause the code to exit, although it's been a while since I last looked at this.

Reply:
I was under the same impression. However, [fail_resource](https://github.com/inspec/inspec/blob/master/lib/inspec/resource.rb#L214) only updates some instance variables of the base class `Inspec.resource(1)`. This is not sufficient to stop the code running in the child class `AzureGenericResources`.